base.cmake

```diff
@@ -237,3 +243,29 @@ MACRO(SETUP_PROJECT_FINALIZE)
   LOGGING_FINALIZE()
 ENDMACRO(SETUP_PROJECT_FINALIZE)
 
+#.rst:
+# .. ifmode:: user
+#
+#   .. command:: COMPUTE_PROJECT_ARGS
+#
+#     Compute the arguments to be passed to command PROJECT.
+#     For instance::
+#
+#       COMPUTE_PROJECT_ARGS(PROJECT_ARGS)
+#       PROJECT(${PROJECT_NAME} ${PROJECT_ARGS})
+#
+#     :param VARIABLE: the variable where to write the result
+#
+MACRO(COMPUTE_PROJECT_ARGS VARIABLE)
+  IF(CMAKE_VERSION VERSION_GREATER "3.0.0")
+    CMAKE_POLICY(SET CMP0048 NEW)
+    SET(${VARIABLE} VERSION ${PROJECT_VERSION_FULL} LANGUAGES CXX)
+
+    # Append description for CMake >= 3.9
+    IF(CMAKE_VERSION VERSION_GREATER "3.9.0")
+      SET(${VARIABLE} ${VARIABLE} DESCRIPTION ${PROJECT_DESCRIPTION})
+    ENDIF(CMAKE_VERSION VERSION_GREATER "3.9.0")
+  ELSE(CMAKE_VERSION VERSION_GREATER "3.0.0")
+    SET(${VARIABLE} CXX)
+  ENDIF(CMAKE_VERSION VERSION_GREATER "3.0.0")
+ENDMACRO(COMPUTE_PROJECT_ARGS)
```

Comment:
This avoids having to write stuff like https://github.com/humanoid-path-planner/hpp-fcl/pull/105/commits/3b473bc13cc1bf2f2b70021e79f76f19cb01887f in every project.

Reply:
That's great but could we introduce `PROJECT_LANGUAGES` (defaults to `CXX`) in order to handle the case when the project does not only uses C++.